### PR TITLE
Ability to customize template

### PR DESCRIPTION
**Pull request by gilbertk88**
Issue #17 that intend to give the plugin an ability to change the look and fee of screenly cast template.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -54,15 +54,14 @@ before_script:
...
       phpunit
       WP_MULTISITE=1 phpunit
     fi
-   -if [ "$TRAVIS_BRANCH" = "master" ] || [ "$TRAVIS_PULL_REQUEST" = "false" ]; 
-        then./bin/push_to_wordpress_org.sh
-        fi
-   - echo $TRAVIS_BRANCH
-   
\ No newline at end of file
+
```
Comment (vpetersson) [inline .travis.yml]: Why did you change these?

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ matrix:
     - php: 7.0
       env: WP_VERSION=latest
     - php: 5.6
-      env: WP_VERSION=3.7
+      env: WP_VERSION=latest
```
Comment (vpetersson) [inline .travis.yml]: This doesn't make sense. Why would we test `WP_VERSION=latest` twice on 5.6?

File: screenly-cast/readmem.txt [hunk 3]
```diff
@@ -0,0 +1,62 @@
+=== Screenly Cast ===
```
Comment (vpetersson) [inline screenly-cast/readmem.txt]: What's up with this file? `readmem.txt`

File: screenly-cast/theme/screenly-cast/assets/js/category_scripts.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            /*
+             * LET'S START REMOVING CHILD BLOCKS THAT ARE OUT OF VIEW
+             */
+            var i = childs.length;
+            for (i; i>0; i--) {
+                var child = childs[i];
+                if (typeof child === 'object') {
+                    var childRect = child.getClientRects()[0];
+
+                    if (childRect.top > article.clientHeight) {
```
Comment (Evgenioni) [inline screenly-cast/theme/screenly-cast/assets/js/category_scripts.js:20]: The error "Cannot read property 'top' of undefine" as the parent (section) has a property display:none. That's why slider doesn't work and revert blank screen.

File: screenly-cast/theme/screenly-cast/assets/js/category_scripts.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	srly_current_post_id=0;
+	srly_post_ceiling=2;
+		
+	if(srly_category_switch_period == undefined){
+		srly_category_switch_period = 5000;
+	}
+	
+	srly_category_manager={
+		rotate_post:function(){	//rotate posts on category
+
```
Comment (Evgenioni) [inline screenly-cast/theme/screenly-cast/assets/js/category_scripts.js:55]: A slider will display incorrectly the posts in category, if their quantity != 3

File: screenly-cast/theme/screenly-cast/category_style.php [hunk 6]
```diff
@@ -0,0 +1,439 @@
+<?php header('Content-type: text/css');
```
Comment (Evgenioni) [inline screenly-cast/theme/screenly-cast/category_style.php:1]: It's a wrong way to generate CSS in this way. Maybe it's better use common CSS-file for render a single post and category's slider, and control view through JS

File: screenly-cast/theme/screenly-cast/style.php [hunk 7]
```diff
@@ -0,0 +1,416 @@
+<?php header('Content-type: text/css'); 
+/**
```
Comment (Evgenioni) [inline screenly-cast/theme/screenly-cast/style.php:2]: It's a wrong way too

File: screenly-cast/inc/functions.php [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+/**
+ * Get custom styles settings, they will be used on the frontend theme. You need to have created them on the admin settings page and format them accordingly.
+ *
+ *
+ * @package ScreenlyCast
+ * @return  string
+ */
+function srly_get_css_settings_from_db() {
+    require_once  dirname(__FILE__).'/../../../../wp-config.php';
```
Comment (Evgenioni) [inline screenly-cast/inc/functions.php:33]: It's non-wordpress way of include wp-config.php. why is it here?

File: screenly-cast/inc/functions.php [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+    $srly_b_strong_font_weight= get_option('srly_b_strong_font_weight');
+	
+    $srly_ul_ol_padding_left= get_option('srly_ul_ol_padding_left');
+    $srly_ul_list_style_type= get_option('srly_ul_list_style_type');
+    $srly_ul_li_padding_left= get_option('srly_ul_li_padding_left');
+	
+    $srly_h3_color = get_option('srly_h3_color');
+    $srly_h3_font_size= get_option('srly_h3_font_size');
+    $srly_h3_font_weight =  get_option('srly_h3_font_weight');
+
```
Comment (Evgenioni) [inline screenly-cast/inc/functions.php:118]: It's nonoptimal mode to store setting in data base. It's better to use serialized array

File: screenly-cast/inc/assets/js/scripts.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+			var attachment = custom_uploader.state().get('selection').first().toJSON();
+			$('#media_image').attr('src', attachment.url);
+			$('#screenly_options_logo').val(attachment.url);
+		})
+		.open();
+	});
+	$('#media_image').attr('src', $('#screenly_options_logo').val());
+	$('.srly_color_picker').wpColorPicker();
+	
+	$('#srly_brand_logo_width').change(function(){
```
Comment (Evgenioni) [inline screenly-cast/inc/assets/js/scripts.js:23]: Is it possible to use default WP's UI controls in settings's page? It's not userfriendly, especially on mobile and tablets

File: screenly-cast/inc/assets/js/scripts.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+	$('#srly_h1_padding').change(function(){
+		$("#srly_h1_padding_slider_output").html($("#srly_h1_padding").val()+"px");
+	});
+	$("#srly_h1_padding_slider_output").html($("#srly_h1_padding").val()+"px");
+	
+	$('#srly_category_switch_period').change(function(){
+		$("#srly_category_switch_period_slider_output").html($("#srly_category_switch_period").val()+" ms");
+	});
+	$("#srly_category_switch_period_slider_output").html($("#srly_category_switch_period").val()+" ms");
+});
```
Comment (Evgenioni) [inline screenly-cast/inc/assets/js/scripts.js:97]: The "Save Settings" button on the form doesn't function by pressing Enter (it's default behavior)

**Review by Evgenioni — COMMENTED** [review]
Hi. I use your plugin, but i have some questions about its work =)

**Comment by vpetersson** [comment]
@Evgenioni Long time overdue comment, but as you've probably noticed, this project has been abandoned. You might however find our [Edge Apps](https://www.screenly.io/developers/edge-apps/) interesting.

**Comment by gilbertk88** [comment]
Hey @vpetersson, hope you are doing well, I just saw the comments, sorry about that. I'd be happy to resolve the issues right away - If you need me to.

**Comment by vpetersson** [comment]
This entire plugin probably requires a big overhaul as a lot of things have changed on WordPress' side.

**Comment by vpetersson** [comment]
Closing out this PR as we've resurrected things here:
https://github.com/Screenly/Screenly-Cast-for-WordPress/pull/25
